#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "../protos/SLRC_supervisor_v1.proto"

WorldInfo {
  contactProperties [
    ContactProperties {
    }
  ]
}
Viewpoint {
  orientation 0.23541535224222226 0.2898666607170327 -0.9276620779860666 1.4381926736405584
  position -1.3783704147400317 -0.4933305467916721 0.3112367344152247
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation -0.62 0 0
  floorSize 2.7 2.7
  floorTileSize 5 5
  floorAppearance Parquetry {
    type "dark strip"
    colorOverride 0 0 0
    textureTransform TextureTransform {
    }
  }
}
DEF box_3 Solid {
  translation -1.698269732338189 -0.9897460069438065 0.024995095
  rotation 0.5874042016905369 -0.5566979501243076 -0.587404201690537 -2.1256505677987034
  children [
    DEF box_3_shape Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.05 0.05 0.05
      }
    }
  ]
  name "solid(4)"
  boundingObject USE box_3_shape
  physics Physics {
    density 100
  }
}
DEF box_2 Solid {
  translation 0.4800000000000835 -1.089999999999917 0.019997488640000012
  rotation -0.048981377085098414 0.055663361689487746 0.9972474190812793 3.0423166331239976e-15
  children [
    DEF box_2_shape Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.04 0.04 0.04
      }
    }
  ]
  name "solid(3)"
  boundingObject USE box_2_shape
  physics Physics {
    density 100
  }
}
DEF box_1 Solid {
  translation 0.4800000000000151 1.1300000000000143 0.014998940519999998
  rotation 0.14010829470710545 0.39464508782250146 -0.908088608238115 2.427674321213902e-16
  children [
    DEF box_1_shape Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        metalness 0
      }
      geometry Box {
        size 0.03 0.03 0.03
      }
    }
  ]
  name "solid(2)"
  boundingObject USE box_1_shape
  physics Physics {
    density 100
  }
}
DEF maze Solid {
  translation -1.93 1.32 -0.00589861
  rotation 1 0 0 1.5708
  scale 0.001 0.001 0.001
  children [
    DEF maze_shape Shape {
      appearance PBRAppearance {
        metalness 0
      }
      geometry Mesh {
        url [
          "../solid bodies/arena/webotsArena.STL"
        ]
      }
      castShadows FALSE
    }
  ]
}
DEF supervisor SLRC_supervisor_v1 {
}
DEF robot Robot {
  translation -1.272527357610555 -1.0596912021672007 0.026892872046686552
  rotation -8.952201414402849e-05 0.00012215467414408737 0.9999999885320222 1.777997874796861
  children [
    SliderJoint {
      jointParameters JointParameters {
        position 0.05499972532000745
      }
      device [
        LinearMotor {
          name "linear_motor"
          maxPosition 0.06
        }
      ]
      endPoint Solid {
        translation 0.054000000135493 5.848377337969168e-12 0.08499972509395282
        rotation 3.086446019308867e-05 -0.9999999988122953 -3.771995000102624e-05 -4.212082238552739e-09
        children [
          Solid {
            translation 0.05 0 0
            children [
              DEF right_arm_slider SliderJoint {
                jointParameters JointParameters {
                  position 1.2482188025878644e-09
                  axis 0 1 0
                }
                device [
                  DEF right_arm LinearMotor {
                    name "right_arm"
                    minPosition -2.3875475041314083e-10
                    maxPosition 0.03
                  }
                ]
                endPoint Solid {
                  translation 0 -0.03999999851302645 -0.015
                  children [
                    DEF leftarm Shape {
                      appearance PBRAppearance {
                        baseColor 0.6 0 0.4
                        metalness 0
                      }
                      geometry Box {
                        size 0.01 0.01 0.04
                      }
                    }
                  ]
                  name "solid(1)"
                  boundingObject USE leftarm
                  physics Physics {
                  }
                }
              }
              DEF left_arm_slider SliderJoint {
                jointParameters JointParameters {
                  position 2.3417251225282437e-12
                  axis 0 -1 0
                }
                device [
                  DEF left_arm LinearMotor {
                    name "left_arm"
                    maxPosition 0.03
                  }
                ]
                endPoint Solid {
                  translation 0 0.039999999997658284 -0.015
                  children [
                    DEF leftarm Shape {
                      appearance PBRAppearance {
                        baseColor 0.6 0 0.4
                        metalness 0
                      }
                      geometry Box {
                        size 0.01 0.01 0.04
                      }
                    }
                  ]
                  boundingObject USE leftarm
                  physics Physics {
                  }
                }
              }
              DEF crossbar Shape {
                appearance PBRAppearance {
                  baseColor 0.2 0.3 0.3
                  metalness 0
                }
                geometry Box {
                  size 0.01 0.1 0.01
                }
              }
            ]
            name "cross bar"
            boundingObject USE crossbar
            physics Physics {
            }
          }
          Solid {
            translation 0.05 0 0
            children [
              SliderJoint {
                jointParameters JointParameters {
                  position -3.5315963817281445e-07
                }
                device [
                  LinearMotor {
                  }
                ]
              }
            ]
          }
          DEF ab_shape Shape {
            appearance PBRAppearance {
              baseColor 0.1 0.5 1
              metalness 0
            }
            geometry Box {
              size 0.1 0.01 0.01
            }
          }
        ]
        name "solid(5)"
        boundingObject USE ab_shape
        physics Physics {
        }
      }
    }
    Solid {
      translation 0 0 0.06
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.6 0.4 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.1
          }
        }
      ]
      name "solid(4)"
    }
    DistanceSensor {
      translation 0.075 0.02 0
      rotation 0 0 1 0.5
      children [
        DEF distant_sensor Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds left"
      boundingObject USE distant_sensor
      physics Physics {
      }
    }
    DEF ds DistanceSensor {
      translation 0.075 -0.02 0
      rotation 0 0 1 -0.5
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds right"
      boundingObject USE distant_sensor
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 115.9082581800221
        axis 0 1 0
        anchor 0.05 0.056 -0.007
      }
      device [
        DEF front_left_wheel RotationalMotor {
          name "front_left_wheel"
        }
      ]
      endPoint Solid {
        translation 0.05 0.056 -0.007
        rotation 0.11708607482569053 0.7027757123548277 -0.701710160398207 2.9081311114559156
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.02
            }
          }
        ]
        boundingObject USE wheel
        physics DEF physics_wheel Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 144.6992467946261
        axis 0 1 0
        anchor 0.05 -0.056 -0.007
      }
      device [
        DEF front_right_wheel RotationalMotor {
          name "front_right_wheel"
        }
      ]
      endPoint Solid {
        translation 0.05000086880316977 -0.05599999334029788 -0.00699539192479185
        rotation -0.9914074504665206 -0.09253344413450469 0.09245987711477947 4.704554730147255
        children [
          USE wheel
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics USE physics_wheel
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 115.90817290069423
        axis 0 1 0
        anchor -0.05 0.056 -0.007
      }
      device [
        DEF left_back_wheel RotationalMotor {
          name "left_back_wheel"
        }
      ]
      endPoint Solid {
        translation -0.04999846598465894 0.05600000146246475 -0.0070044621716677125
        rotation 0.11715803508804275 0.7025166069422234 -0.7019575569681505 2.9081558302044965
        children [
          USE wheel
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics USE physics_wheel
      }
    }
    DEF hing HingeJoint {
      jointParameters HingeJointParameters {
        position 144.6992589088167
        axis 0 1 0
        anchor -0.05 -0.056 -0.007
      }
      device [
        RotationalMotor {
          name "right_back_wheel"
        }
      ]
      endPoint Solid {
        translation -0.049999128891659264 -0.05599999855777699 -0.006995362101142229
        rotation -0.9914062887068926 -0.09253971062963685 0.09246606225534162 4.704554384376439
        children [
          USE wheel
        ]
        name "solid(3)"
        boundingObject USE wheel
        physics USE physics_wheel
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 0 0.2 0.3
        metalness 0
      }
      geometry Box {
        size 0.15 0.1 0.03
      }
    }
  ]
  name "myrobot"
  boundingObject USE body
  physics Physics {
  }
  controller "my_keyboad_controller"
}
